src/edu/ucsb/nceas/metacat/dataone/D1AuthHelper.java

```diff
@@ -726,54 +715,72 @@ public boolean isLocalCNAdmin(Session session) throws ServiceFailure {
+     * Checks Metacat properties representing the local Node document for matching Node.subjects.
+     * The NodeType parameter can be set to limit this authorization check if needed.
+     *
      * @param session
-     * @param nodeType  
+     * @param nodeType
      * @return
      * @throws ServiceFailure
      */
     protected boolean isLocalNodeAdmin(Session session, NodeType nodeType) throws ServiceFailure {
-
         boolean allowed = false;
-        
-        // must have a session in order to check admin 
-        if (session == null) {
-           logMetacat.debug("In isLocalNodeAdmin(), session is null ");
-           return false;
+        // Session must be valid in order to check for authorization
+        String sessionSubjectValue;
+        try {
+            sessionSubjectValue = checkSessionAndGetSubjectValue(session);
+        } catch (NotAuthorized na) {
+            return allowed;
         }
-        
-        logMetacat.debug("In isLocalNodeAdmin(), MN authorization for the user " +
-             session.getSubject().getValue());
-        
+        logMetacat.debug("D1AuthHelper.isLocalNodeAdmin(), MN authorization for the user: "
+            + sessionSubjectValue);
+
         Node node = MNodeService.getInstance(request).getCapabilities();
         NodeReference nodeReference = node.getIdentifier();
-        logMetacat.debug("In isLocalNodeAdmin(), Node reference is: " + nodeReference.getValue());
-        
+        logMetacat.debug(
+            "D1AuthHelper.isLocalNodeAdmin(), Node reference is: " + nodeReference.getValue());
+
         Set<Subject> sessionSubjects = AuthUtils.authorizedClientSubjects(session);
-        
+
+        // First, check for a local node subject
         if (nodeType == null || node.getType() == nodeType) {
             List<Subject> nodeSubjects = node.getSubjectList();
             if (sessionSubjects != null) {
-              outer:
+                outer:
                 for (Subject subject : sessionSubjects) {
-                    // check if the session subject is in the node subject list
+                    // Check if the session subject is in the node subject list
                     for (Subject nodeSubject : nodeSubjects) {
-                        logMetacat.debug("In isLocalNodeAdmin(), comparing subjects: " +
-                            nodeSubject.getValue() + " and the user" + subject.getValue());
-                        if ( nodeSubject.equals(subject) ) {
-                            allowed = true; // subject of session == this node's subect
+                        logMetacat.debug("D1AuthHelper.isLocalNodeAdmin(), comparing node subject: "
+                            + nodeSubject.getValue() + " and the session user: "
+                            + subject.getValue());
+                        if (nodeSubject.equals(subject)) {
+                            allowed = true; // subject of session == this node's subject
                             break outer;
```

Comment:
delete these 2 lines. replace with:

```java
    logMetacat.debug("Found " + subject.getValue() + " in the node subject list. Returning true");
    return true;
```

(no need to include D1AuthHelper.isLocalNodeAdmin() - this is already added to the end of the log line by log4j)